docker/run.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+set -e
+
+BUILDER_VOLUME="$(dirname $PWD)"
+CACHER_VOLUME="$HOME/apt_cacher_mnt"
+IMG="derivative-maker/derivative-maker-docker"
+USER="user"
+LOG_DIR="$PWD/logs"
+[ -d ${LOG_DIR} ] || mkdir -p ${LOG_DIR}
+
+sudo modprobe -a loop dm_mod
+
+sudo docker run --name derivative-maker-docker -it --rm --privileged \
+	--env "TAG=17.4.0.3-developers-only" \
+ 	--env 'flavor_meta_packages_to_install=' \
+	--env 'install_package_list=' \
+	--env 'DERIVATIVE_APT_REPOSITORY_OPTS=' \
+	--volume ${BUILDER_VOLUME}:/home/user \
```

Comment:
This is problematic, because it results in the derivative-maker repo being mounted to `/home/user` in the container. There is other code that assumes the repo will be at `/home/user/derivative-maker` though. For me, this resulted in the following error:

```
+ source /home/user/derivative-maker/help-steps/pre
/usr/bin/dm-reprepro-wrapper: line 28: /home/user/derivative-maker/help-steps/pre: Not a directory
```

To solve this, let's put the derivative-maker repo in `/home/user/derivative-maker`. This will result in `derivative-binary` being left in an ephemeral location, but we can use some a symlink to fix that in start_build.sh.

```suggestion
	--volume ${BUILDER_VOLUME}:/home/user/derivative-maker \
```

Reply:
Yes, that is caused by `dm-prepare-release` which expects 
```
"$HOME/derivative-maker/help-steps/pre"
"$HOME/derivative-maker/help-steps/variables"
"$HOME/derivative-maker/help-steps/colors"
```

The way I solved this in the other repo is by separately cloning derivative-maker/help-steps and each tag by tag id.
```
[ -d ~/derivative-maker ] || { mkdir -p ~/derivative-maker; \
cd ~/derivative-maker; git init -b master; \
git remote add -f origin ${GIT_URL}; \
git config core.sparseCheckout true; \
cat > .git/info/sparse-checkout << EOF
help-steps/pre
help-steps/colors
help-steps/variables
EOF
```
This allows tags to be cloned separately into folders named by the individual tag id. When restarting the container, based on the tag that is passed to `start_build.sh`, the script will either jump into an existing folder (if that tag's already been cloned) and do git pull etc. or create a new one.
Basically, I wanted to have everything separate, and simplify repeated use of the container.

In this case, the source code is already manually cloned, so mounting the entire thing as `$BUILDER_VOLUME` made sense. But how do you automate it with container restarts and new tags in mind?